syntax = "proto3";

package enrichment;

// Service definition for map polygon enrichment
service EnrichmentService {
  rpc EnrichPolygon (PolygonRequest) returns (EnrichmentResponse);
}

// Request message containing polygon coordinates
message PolygonRequest {
  repeated Coordinate coordinates = 1;
}

// Coordinate point (lat, lng)
message Coordinate {
  double lat = 1;
  double lng = 2;
}

// Business/POI information
message Business {
  string name = 1;
  double lat = 2;
  double lng = 3;
  string type = 4;
  string address = 5;
  string phone = 6;
  string website = 7;
  string email = 8;
}

// Response with enriched data
message EnrichmentResponse {
  double area_km2 = 1;
  int32 estimated_population = 2;
  string region_type = 3;
  repeated string nearby_features = 4;
  repeated Business businesses = 5;
}
